invenio_db/utils.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2017 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+# from .signals import secret_key_changed
+
+"""Invenio-DB utility functions."""
+
+from flask import current_app
+from sqlalchemy.engine import reflection
+
+from invenio_db import db
+
+
+def rebuild_encrypted_properties(old_key, model, properties):
+    """Rebuild a model's EncryptedType properties when the SECRET_KEY is changed.
+
+    :param old_key: old SECRET_KEY.
+    :param model: the affected db model.
+    :param properties: list of properties to rebuild.
+    """
+    inspector = reflection.Inspector.from_engine(db.engine)
+    primary_key_names = inspector.get_primary_keys(model.__tablename__)
+
+    new_secret_key = current_app.secret_key
+    db.session.expunge_all()
+    with db.session.begin_nested():
+        current_app.secret_key = old_key
```

Comment:
please make sure that the secret key is restored in case of exception (missing `finally:`)